File: dailyQuestion3499.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int max(int a, int b) {
    return a > b ? a : b;
}

int maxActiveSectionsAfterTrade(char* s) {

    int ones = 0;
    int n = strlen(s);

    for (int i = 0; i < n; i++) {
        if (s[i] == '1')
            ones++;
    }

    char *t = (char *)malloc(n + 3);

    t[0] = '1';
    strcpy(t + 1, s);
    t[n + 1] = '1';
    t[n + 2] = '\0';

    int len = n + 2;

    char *type = (char *)malloc((len + 1) * sizeof(char));
    int *cnt = (int *)malloc((len + 1) * sizeof(int));

    int runs = 0;

    for (int i = 0; i < len;) {

        int j = i;

        while (j < len && t[j] == t[i])
            j++;

        type[runs] = t[i];
        cnt[runs] = j - i;
        runs++;

        i = j;
    }

    int ans = ones;

    for (int i = 1; i + 1 < runs; i++) {

        if (type[i] == '1') {

            int gain = cnt[i - 1] + cnt[i + 1];

            ans = max(ans, ones + gain);
        }
    }

    free(t);
    free(type);
    free(cnt);

    return ans;
}

int main() {

    char s[] = "11000111";

    int result = maxActiveSectionsAfterTrade(s);

    printf("Maximum Active Sections After Trade: %d\n", result);

    return 0;
}

File: dailyQuestion3499.java
import java.util.*;

class Main {

    public static void main(String[] args) {

        String s = "11000111";

        Solution obj = new Solution();

        int result = obj.maxActiveSectionsAfterTrade(s);

        System.out.println("Maximum Active Sections After Trade: " + result);
    }
}

class Solution {

    public int maxActiveSectionsAfterTrade(String s) {

        int ones = 0;

        for (char c : s.toCharArray()) {
            if (c == '1') {
                ones++;
            }
        }

        String t = "1" + s + "1";

        ArrayList<int[]> runs = new ArrayList<>();

        for (int i = 0; i < t.length();) {

            int j = i;

            while (j < t.length() && t.charAt(j) == t.charAt(i)) {
                j++;
            }

            runs.add(new int[]{t.charAt(i), j - i});

            i = j;
        }

        int ans = ones;

        for (int i = 1; i + 1 < runs.size(); i++) {

            if (runs.get(i)[0] == '1') {

                int gain = runs.get(i - 1)[1] + runs.get(i + 1)[1];

                ans = Math.max(ans, ones + gain);
            }
        }

        return ans;
    }
}